TG-3 TG-9 Avance...

evelynda1985 opened the pull request:
Se comento el metodo main. 
Se paso la clase Ingrediente a abstracta, se creo el metodo tamano como abstracto y se crearon 6 sub clases (Liquidos, Granos, Carnes, VerdurasYFrutas, Condimentos) de la super clase Ingrediente.

Commit:
TG-3 TG-9 Avance...
Se comento el metodo main. 
Se paso la clase Ingrediente a abstracta, se creo el metodo tamano como abstracto y se crearon 6 sub clases (Liquidos, Granos, Carnes, VerdurasYFrutas, Condimentos) de la super clase Ingrediente.

jramirezl commented on src/receta/controller/Main.java:
```diff
@@ -20,7 +20,7 @@
   
 
-    public static void main(String[] args) {
+    /*public static void main(String[] args) {
```
Comenta todo el main

evelynda1985 replied:
listo teacher Juan :p

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
-public class Ingrediente {
+
+public abstract class Ingrediente {
 
```
Varias cosas:
LO bueno 👍  : 
Hiciste bien la relación entre las clases.

LO malo 👍 

Recuerda poner siempre cada clase en un archivo .java distinto. Hay casos execpionales. En este caso puntual. Hay que crear la subclase Líquidos en otro archivo java como los otros.

Se deben extraer nuevos atributos que tu creas. Por ejemplo la leche no tiene tamano o si ? eso te lo dejo :P 
La idea es tratar de identificar atributos que hagan diferente a esos objetos y dejar pocos atributos comunes.

El constructor no te preocupes lo puedes comentar por el momento. Por eso recomende comentar todo el main. El tema del constructor lo podemos dejar quieto mietras hacemos refactor de la nueva estructura. Si tienes mas dudas charlamos :)

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
 }
+
+class Verduras extends Ingrediente{
+
```
Crear en un archivo .java distinto

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
+    }
+}
+
+class Frutas extends Ingrediente{
```
Crear en un archivo .java distinto

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
+
+class Frutas extends Ingrediente{
+
+    public Frutas(String nombreIng, int cantIng, String tipoIng, String tamanoIng) {
```
Crear en un archivo .java distinto

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
-        this.tipo = tipo;
+class Granos extends Ingrediente{
+
+    public Granos(String nombreIng, int cantIng, String tipoIng, String tamanoIng) {
```
Crear en un archivo .java distinto

jramirezl commented on src/receta/Ingrediente.java:
```diff
@@ -9,55 +9,115 @@
+
+class Carnes extends Ingrediente{
+
+    public Carnes(String nombreIng, int cantIng, String tipoIng, String tamanoIng) {
```
Crear en un archivo .java distinto

Commit:
TG-3  TG-22 
Se creo el constructor en Receta con parametros, el cual recibe la cantidad de ingredientes.
